Tests/MarkdownTests/Inline Nodes/LinkTests.swift

```diff
@@ -48,4 +48,29 @@ class LinkTests: XCTestCase {
         link.destination = "test.example.com"
         XCTAssertFalse(link.isAutolink)
     }
+    
+    func testTitleLink() throws {
+        let markdown = #"""
+        [Example](example.com "The example title")
+        [Example2](example2.com)
+        """#
+        
+        let document = Document(parsing: markdown)
+        XCTAssertEqual(document.childCount, 1)
+        let paragraph = try XCTUnwrap(document.child(at: 0) as? Paragraph)
+        XCTAssertEqual(paragraph.childCount, 3)
+
+        XCTAssertTrue(paragraph.child(at: 1) is SoftBreak)
+        let linkWithTitle = try XCTUnwrap(paragraph.child(at: 0) as? Link)
+        let linkWithoutTitle = try XCTUnwrap(paragraph.child(at: 2) as? Link)
+
+        
+        XCTAssertEqual(try XCTUnwrap(linkWithTitle.child(at: 0) as? Text).string, "Example")
+        XCTAssertEqual(linkWithTitle.destination, "example.com")
+        XCTAssertEqual(linkWithTitle.title, "The example title")
+        
+        XCTAssertEqual(try XCTUnwrap(linkWithoutTitle.child(at: 0) as? Text).string, "Example2")
+        XCTAssertEqual(linkWithoutTitle.destination, "example2.com")
+        XCTAssertEqual(linkWithoutTitle.title, "")
```

Comment:
Actually, I expect this to be
```suggestion
        XCTAssertEqual(linkWithoutTitle.title, nil)
```

But that's how `destination` and `title` is computed.

> `cmark_node_get_title` and `cmark_node_get_url` will use `cmark_chunk_to_cstr` which will produce an empty string for a nil input.

We can fix it by updating the getter logic on `Link.title`. The question is should we update the destination's getter accordingly? If so, it may break the downstream since it was a public symbol.

```
/// The link's title.
var title: String? {
    get {
        guard case let .link(_, title) = _data.raw.markup.data else {
            fatalError("\(self) markup wrapped unexpected \(_data.raw)")
        }
		if let t = title, t.isEmpty {
			return nil
		} else {
	        return title
		}
    }
	...
}
```

Reply:
Alternatively we could check if the `cmark_node_get_title` value was empty in CommonMarkConverter.swift before passing the title to the `MarkupConversion` initializer

Reply:
But I agree that I would also expect `nil` here

Reply:
Both implementations are fine for me. The only question is should we sync such logic to `Link.destination`?

Reply:
> Alternatively we could check if the `cmark_node_get_title` value was empty in CommonMarkConverter.swift before passing the title to the `MarkupConversion` initializer

+1 for this since we've already similar operations for `language`.

https://github.com/apple/swift-markdown/blob/07da205e1fc488d88609fea5da1ca227efc6c000/Sources/Markdown/Parser/CommonMarkConverter.swift#L329

Reply:
Done with 2766399